### PR TITLE
Add compatibility with php8.2

### PR DESCRIPTION
**Pull request by misterxan**
- I updated the necessary libraries for PHP 8.2.
- I've eliminated errors related to Undefined key
- I made changes to the logic in `src/Optimizer/FullNameOptimizer.php` in the `removeNamespaceFromString` method, assuming that `$currentNameSpacePaths` would, in one way or another, have a larger dimension than `$dataNameSpacePaths`. Previously, it was the other way around, which caused a warning with an undefined key 2 when I ran tests.
- Fix in another place after phpstan and php-cs-fixer 
- It's possible that I may have inadvertently violated backward compatibility with PHP 7.4 in some places.

The question is whether to keep `$method->setAccessible(true);` in tests since it has no effect starting from 8.1. However, compatibility with 8.0 will be lost, so I think it's worth not removing it.

### CODE REVIEW

**Comment by misterxan** [comment]
@nick-zh, please review.

File: src/Registry/SchemaRegistry.php [hunk 1]
```diff
@@ -135,12 +135,13 @@ private function registerSchemaFile(\SplFileInfo $fileInfo): void
         }
 
         $schemaData = json_decode($fileContent, true, JSON_THROW_ON_ERROR);
-        $namespace = (string) $schemaData['namespace'];
 
         if (null === $schemaData) {
             throw new SchemaRegistryException(sprintf(SchemaRegistryException::FILE_INVALID, $fileName));
         }
 
+        $namespace = array_key_exists("namespace", $schemaData) ? (string) $schemaData['namespace'] : '';
```
Comment (nick-zh) [inline src/Registry/SchemaRegistry.php:143]:
```suggestion
        $namespace = array_key_exists('namespace', $schemaData) ? (string) $schemaData['namespace'] : '';
```

**Review by nick-zh — CHANGES_REQUESTED** [review]
Many thanks for this contribution! :muscle:

File: docker/dev/php/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM php:7.4-cli-alpine3.15
+FROM php:8.2-cli-alpine3.16
```
Comment (nick-zh) [inline docker/dev/php/Dockerfile:1]:
```suggestion
FROM php:7.4-cli-alpine3.16
```
i kept 7.4 on purpose to see if backward compatibility was broken somewhere :smile:

File: src/Registry/SchemaRegistry.php (hunk 1 above)
Comment (misterxan) [reply src/Registry/SchemaRegistry.php]: Yeah, good point, I fixed it)

File: docker/dev/php/Dockerfile (hunk 2 above)
Comment (misterxan) [reply docker/dev/php/Dockerfile]: returned it back

**Comment by nick-zh** [comment]
@misterxan thx a lot again, released as `v3.0.0-beta1` 🎉

**Pull request merged**